extjsdk/README.md

```diff
@@ -25,11 +25,27 @@
 	*	[com.fasterxml.jackson.core:jackson-databind Version 2.9.3](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind/2.9.3)
 
 
+## Program Flow
+1.	A new ExtensionWebSocketClient is created.
+2.	Setup and add any handlers through `client.set<type>Handler`. See the [Receiving Messages](#handler) section for more information on what they do. The handlers can be set and reset at any point in time, but you should do so here, or in step 6 through the configuration handler if its setup depends on the contents configuration document.
+3.	Call `client.initiateFullConnection`. Note that all the following steps happen asynchronously relative to the caller, but in the order specified relative to each other.
+4.	The WebSocket connection either succeeds or fails. If it succeeds, the Future returned by `client.getWebSocketFuture()` completes as true and the program continues. If it fails, the CompletableFuture returned from the call in step 3 will complete as false and you should exit or restart at step 3.
+5.	The authentication either succeeds or fails. If it succeeds, the Future returned by `client.getAuthenticationFuture()` completes as true. If it fails, the Future from step 3 completes as false and you should exit or call `client.authenticate; client.connectToSource` or `client.initiateFullConnection` to retry step 5. Regardless of success or failure, authHandler is then called. If the authentication succeeded, the program continues.
+6.	The source connection either succeeds or fails. If it succeeds, the Future returned by `client.getSourceConnectionFuture()` completes as true as does the Future from step 3, configHandler is called, and the program continues. If it fails, the Future from step 3 completes as false, httpHandler is called, and then you should exit or call `client.connectToSource` or `client.initiateFullConnection` to retry step 6.
+7.	The source is now connected. Query, Publish, and reconnection messages will be received by their respective handlers. Notification and Query responses/errors can be sent. If a reconnect message is received then the program returns to just before step 6 (but will not proceed to 6 unless you tell it to), reconnectHandler is called, and if the Client's autoReconnect option has been set to true it will proceed to step 6.
```

Comment:
will not proceed unless you tell it to == you have setAutoReconnect(true)?  That might be better to say.  "You tell it to" is a bit colloquial in imprecise for here.